scrape/manager.go

```diff
@@ -264,7 +265,7 @@ func (m *Manager) ApplyConfig(cfg *config.Config) error {
 			sp.stop()
 			delete(m.scrapePools, name)
 		} else if !reflect.DeepEqual(sp.config, cfg) {
-			err := sp.reload(cfg)
+			err := sp.reload(cfg, m.opts.UseSharedInterner)
```

Comment:
nit: flags can never change between reloads, maybe store the option in the struct instead?